webapp/app.py

```diff
@@ -1,172 +1,148 @@
-# Standard library
-import os
-
 # Packages
 import flask
 from canonicalwebteam.flask_base.app import FlaskBase
+from canonicalwebteam.http import CachedSession
 
 # Local
-from webapp.api import get, get_device_information_by_hardware_id
-
+from webapp.api import CertificationAPI
 
-dir_path = os.path.dirname(os.path.abspath(__file__))
-app_dir = os.path.dirname(dir_path)
-templates_dir = os.path.join(app_dir, "templates")
 
 app = FlaskBase(
     __name__,
     "certification.ubuntu.com",
-    template_folder=templates_dir,
+    template_folder="../templates",
     static_folder="../static",
     template_404="404.html",
     template_500="500.html",
 )
 
+api = CertificationAPI(
+    base_url="https://certification.canonical.com/api/v1",
+    session=CachedSession(),
+)
+
 
-@app.route("/hardware/<hardware_id>")
-def hardware(hardware_id):
-    modelinfo_data = get_device_information_by_hardware_id(hardware_id)
+@app.route("/hardware/<canonical_id>")
+def hardware(canonical_id):
+    model_info = api.certifiedmodel(canonical_id)
+    model_devices = api.certifiedmodeldevices(
+        canonical_id=canonical_id, limit="0"
+    )["objects"]
+    model_releases = api.certifiedmodeldetails(
+        canonical_id=canonical_id, limit="0"
+    )["objects"]
 
     hardware_details = {}
-    for component in modelinfo_data.get("hardware_details"):
-        category = component.get("category")
-        if category != "BIOS":
-            category = category.lower()
-        if category not in hardware_details:
-            hardware_details[category] = []
 
-        hardware_info = {
-            "name": f"{component.get('make')} {component.get('name')}",
-            "bus": component.get("bus"),
-            "identifier": component.get("identifier"),
+    for device in model_devices:
+        device_info = {
+            "name": f"{device['make']} {device['name']}",
+            "bus": device["bus"],
+            "identifier": device["identifier"],
         }
 
-        hardware_details[category].append(hardware_info)
+        category = device["category"]
+        if category != "BIOS":
+            category = category.lower()
+
+        if category in hardware_details:
+            hardware_details[category].append(device_info)
+        else:
+            hardware_details[category] = [device_info]
 
     release_details = {"components": {}, "releases": []}
-    for release in modelinfo_data.get("release_details"):
-        release_version = release["certified_release"]
-        release_name = (
-            "Ubuntu "
-            f"{release_version} "
-            f"{ '64 Bit' if release['architecture'] == 'amd64' else '32 Bit'}"
-        )
+
+    for model_release in model_releases:
+        ubuntu_version = model_release["certified_release"]
+        arch = ""
```

Comment:
We could `arch = "32 Bit"` here and remove the `else`

Reply:
Yeah I considered that, but I think this version, although it's 2 lines longer, is a bit more logically straightforward which I think is more important.